src/components/MainScreen/MainScreen.tsx

```diff
@@ -1,39 +1,44 @@
-import { useEffect, useState, useTransition } from "react";
+import { useCallback, useEffect, useState } from "react";
 import classes from "./MainScreen.module.scss";
 import { ChuckJoke } from "../../types/ChuckJoke";
 import { jokesApi } from "../../api/jokesApi";
 import Image from "../Image/Image";
 import CategorySelect from "../CategorySelect/CategorySelect";
 import AlertSnackbar from "../AlertSnackbar/AlertSnackbar";
-import { AxiosError } from "axios";
+import { AxiosError, AxiosResponse } from "axios";
 import InputForm from "../InputForm/InputForm";
 import LanguagePicker from "../LanguagePicker/LanguagePicker";
+import SaveJokes from "../SaveJokes/SaveJokes";
 import { useTranslation } from "react-i18next";
 
 const MainScreen = () => {
   const [joke, setJoke] = useState<ChuckJoke>();
   const [errorMessage, setErrorMessage] = useState<string>();
   const [category, setCategory] = useState("");
   const [isError, setIsError] = useState(false);
-  const [customName, setCustomName] = useState<string|undefined>();
+  const [customName, setCustomName] = useState<string | undefined>();
+  const [isChuckImage, setIsChuckImage] = useState(true)
+
+  const { t } = useTranslation();
+
+  const fetchJokes = useCallback(async (name?: string) => {
+    try {
+      const axiosResponse = await jokesApi.fetchRandomJokes(category, name) as AxiosResponse<ChuckJoke>
+      setIsError(false);
+      setJoke(axiosResponse.data)
+    } catch (error) {
+      const errorResponse = error as AxiosError
+      setIsError(true)
+      setErrorMessage(errorResponse.message)
+    }
+  }, [category]);
 
   useEffect(() => {
     const getJokes = async () => {
```

Comment:
since you used useCallback you don't need to initialize new function `fetchJokes` will be just fine